src/scripts/ordo_report_mapping_annotations/report_mapping_annotations.py

```diff
@@ -0,0 +1,144 @@
+    """Query ontology using SPARQL query file"""
+    # Basic vars
+    results_filename = 'results.csv'
+    command_save_filename = 'command.sh'
+    results_dirname = os.path.basename(onto_path).replace('/', '-').replace('.', '-') + \
+        '__' + os.path.basename(query_path).replace('.', '-')
+    results_dirpath = os.path.join(CACHE_DIR, 'robot', results_dirname)
+    command_save_path = os.path.join(results_dirpath, command_save_filename)
+    results_path = os.path.join(results_dirpath, results_filename)
+
+    # Use cache if exists and desired, else run and cache
+    if not use_cache or (use_cache and not os.path.exists(results_path)):
+        command_str = f'robot query --input {onto_path} --query {query_path} {results_path}'
+        os.makedirs(results_dirpath, exist_ok=True)
+        if not os.path.exists(results_path):
+            with open(command_save_path, 'w') as f:
+                f.write(command_str)
+        subprocess.run(command_str.split())
+
+    # Read results and return
+    try:
+        df = pd.read_csv(results_path).fillna('')
+    except pd.errors.EmptyDataError as err:
+        os.remove(results_path)  # done so that it doesn't read this from cache
+        raise err
+
+    return df
+
+
+def get_ordo_mappings_report(
+    onto_path: str, query_path: str, mapping_meanings_path: str, use_cache=False
+) -> Dict:
+    """Get ordo mappings
+    :return Dict:Get map of unique ORDO mapping strings to relationship CURIEs """
+    # Get mapping code meanings
+    code_meanings_df: pd.DataFrame = pd.read_csv(mapping_meanings_path)
+    code_meanings = {}
+    for index, row in code_meanings_df.iterrows():
+        code_meanings[row['ordo_mapping_code']] = row['relationship_curie']
+
+    # Get current mapping annotations
+    df: pd.DataFrame = sparql_file_query__via_robot(onto_path, query_path, use_cache)
+    df['cls'] = df['cls'].apply(lambda uri: uri.split('_')[-1])
+    df = df.rename(columns={'cls': 'class_id'})
+
+    # Collect annotation info
+    # - Get classes
+    d = {}
+    for index, row in df.iterrows():
+        map_str: str = row['mapping_precision_string']
+        if map_str not in d:
+            d[map_str] = {'classes': []}
+        d[map_str]['classes'].append(row['class_id'])
+    # - Get n classes
+    d = {
+        k: {
+            'classes': v['classes'],
+            'n': len(v['classes'])
+        }
+        for k, v in d.items()
+    }
+    # - Collate info
+    d2 = {}
+    #  d2 keys: unique ORDO mapping codes
+    #  d2 values: i. assumed formalized mapping predicate, ii. number of mappings, iii. list of Orphanet class IDs that
+    #  ...are the subject of this predicate, iv. unique mapping strings.
+    for k, v in d.items():
+        code: str = k.split('(')[0].replace('-', '').strip()
+        if code not in d2:
+            d2[code] = {'classes': [], 'n': 0, 'unique_mapping_strings': [], 'relationship_curie': code_meanings[code]}
```

Comment:
@matentzn I don't know if I mentioned it on the last call, but this Python script produces a structured report which shows which classes have certain mapping annotations on them. Perhaps it could be useful for human review. But part of that report, in its `dictionary` form (namely the `unique_mapping_strings` and `relationship_curie`, is used by another Python script which instantiates a Jinja template.